rfc-142-split-database-instances.md

```diff
@@ -0,0 +1,50 @@
+
+## Naming
+
+The word "database" is often interchangeably used to describe a collection of tables, and an instance of (e.g.) Postgres server running somewhere. For the purposes of this RFC, "database" refers to the collection of tables, schema, etc.. We'll use "instance" to refer to the runtime that hosts the database.
+
+## Problem
+
+Having a single instance hosting multiple databases has a number of drawbacks:
+
+1. Major upgrades must be applied to all databases in one go, as it's the instance we're upgrading, not the database. Major upgrades are considered risky because they contain potentially breaking, and backward-incompatible changes. This means major upgrades are scarier, so we're less likely to do them until we're forced. They also require coordination across disparate teams so everyone's confident their apps will survive the upgrade.
+
+2. Problems in one database are likely to impact other unrelated services. For example an app with unexpectedly high load on its database could max-out the instance, causing performance degradation in other databases and their apps.
+
+3. Related to [2], it's significantly harder to size instance resources. Disk is fairly straightforward, but it's harder to understand CPU/RAM/IOPS patterns. We're also likely to over-size resources to reduce the likelihood large-scale problems in [2].
+
+4. There is no clear owner of the instances. Each app is owned by a single team, and each database should be accessed by exactly one app. However many apps with different owners reside on the single instance. Sizing and managing databases is an app-level responsibility, so it doesn't fit with the remit of Platform Reliability/Replatforming/Platform Health. Having multiple databases on a single instance 
+
+This is coming up now because our versions of MySQL and Postgres are approaching End of Life (notice for [MySQL](https://forums.aws.amazon.com/ann.jspa?annID=8790) and [Postgres](https://forums.aws.amazon.com/ann.jspa?annID=8499)), forcing us to plan major upgrades for both over the next few months.
+
+## Proposal
+
+We propose that:
+
+1. All new long-lived databases are created in their own individual RDS instances.
+2. Existing MySQL and Postgres databases are migrated out of the central instance to individual managed-service instances as part of the upcoming major upgrade cycle.
+3. Existing Redis and Mongo databases are migrated out of their central instances in-line with team priorities, and prior to any major upgrade.
+
+
+### Consequences
+
+- Creating more RDS instances means that application-owning teams will need to monitor and consider database upgrades as part of their day-to-day work. We can ease this burden through monitoring & reporting, which may be considered by a future Platform Reliability team.
+
+- Zero-downtime migrations take extra planning and effort. Services such as AWS DB Migration Service ease some of this complexity.
```

Comment:
This one seems a little out of the blue - perhaps it's: There may need to be some service unavailability to apply these principles - however this would happen regardless by AWS ending support (and AWS DB Migration service can help) ?

Reply:
I've reworded to make the point that we'd have to upgrade anyway. I thought it was worth calling out because the additional complexity of splitting the databases could increase downtime (or at least force people to think about it), but happy to remove if the reworded one doesn't fit with the rest of the RFC